### PR TITLE
Task/des 1291: Move publication filters to backend

### PR DESCRIPTION
**Pull request by jarosenb**
The old implementation was bad at handling publications >100 places deep in the list. Now the filters manipulate the state and filtering is done on the backend. 
Other changes:
- renames the `files-listing-public` component to `publications-listing` for legibility
- deprecate the `publicDataSearch` state and use `publicData` for both search and listings.
- Change `project` and `project.value` fields in `IndexedPublication` from nested to object fields so that the publication type can be filtered on without having to execute a 3-level nested query

**Requires `./manage.py swap-reindex --index=publications`**

### CODE REVIEW

File: designsafe/static/scripts/data-depot/components/data-depot-listing/publications-listing.template.html [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+    <p ng-if="$ctrl.$stateParams.query_string">{{$ctrl.browser.listing.children.length}}<span ng-if="$ctrl.browser.listing.children.length >= 100">+</span> Results found for <b>{{$ctrl.$stateParams.query_string}}</b></p>
+    <p><b>Publication Type</b> <a ng-if=$ctrl.$stateParams.typeFilters ng-click="$ctrl.clearTypeFilters()">Clear</a></p>
+    <label class="checkbox-inline"><input type="checkbox" ng-checked="$ctrl.checkTypeFilterSelected('experimental')" ng-click="$ctrl.onTypeFilterSelect('experimental')"  >Experimental</label>
+    <label class="checkbox-inline"><input type="checkbox" ng-checked="$ctrl.checkTypeFilterSelected('simulation')" ng-click="$ctrl.onTypeFilterSelect('simulation')"  >Simulation</label>
+    <label class="checkbox-inline"><input type="checkbox" ng-checked="$ctrl.checkTypeFilterSelected('hybrid_simulation')" ng-click="$ctrl.onTypeFilterSelect('hybrid_simulation')">Hybrid Simulation</label>
+    <label class="checkbox-inline"><input type="checkbox" ng-checked="$ctrl.checkTypeFilterSelected('field_recon')" ng-click="$ctrl.onTypeFilterSelect('field_recon')">Field Recon</label>
```
Comment (Strmiska) [inline designsafe/static/scripts/data-depot/components/data-depot-listing/publications-listing.template.html]: This should be "Field Research" (this was changed recently)

**Comment by Strmiska** [comment]
I made a pr to add the "Field Research" tag to the selectable options but I'm going to close it since this covers it. I had a section here though that I wanted to show you:
https://github.com/DesignSafe-CI/portal/pull/645/files#diff-259dfb1e5433661f3aba6c2d5b8551dcR192

I think the project type tags (next to the title in the publication listing) need to be updated for Field Research and Hybrid Simulation.

File: designsafe/static/scripts/ng-designsafe/services/data-browser-service.js [hunk 2]
```diff
@@ -1996,10 +1989,12 @@ export function DataBrowserService($rootScope, $http, $q, $uibModal,
     }
     currentState.page += 1;
     currentState.loadingMore = true;
+    console.log(params.typeFilters)
```
Comment (jarosenb) [inline designsafe/static/scripts/ng-designsafe/services/data-browser-service.js]: Note to self: gotta remove this console.log

File: designsafe/apps/api/search/searchmanager/publications.py [hunk 3]
```diff
@@ -28,10 +28,19 @@ def __init__(self, request=None, **kwargs):
         super(PublicationsSearchManager, self).__init__(
             IndexedPublication, Search())
 
-    def construct_query(self, system=None, file_path=None):
+    def construct_query(self, system=None, file_path=None, **kwargs):
 
         published_index_name = Index('des-publications').get_alias().keys()[0]
         legacy_index_name = Index('des-publications_legacy').get_alias().keys()[0]
+        logger.debug(kwargs)
```
Comment (jarosenb) [inline designsafe/apps/api/search/searchmanager/publications.py]: remove this logging statement

**Pull request merged**